test/filters/test_convolution.cpp

```diff
@@ -3155,6 +3158,268 @@ main (int argc, char** argv)
+  PointCloud<RGB>::Ptr input (new PointCloud<RGB> ());
+  input->width = 48;
+  input->height = 48;
+  input->resize(input->width * input->height);
+  for (std::uint32_t r = 0; r < input->height; r++)
+    for (std::uint32_t c = 0; c < input->width; c++)
+    {
+      float x1 = -2.0f + (4.0f / (float)input->width) * (float)c;
+      float y1 = -2.0f + (4.0f / (float)input->height) * (float)r;
+      float x2 = -M_PI + (2.0f * M_PI / (float)input->width) * (float)c;
+      float y2 = -2.0f + (4.0f / (float)input->height) * (float)r;
+      float z = x1 * exp(-(x1 * x1 + y1 * y1)) * 2.5f + sin(x2) * sin(y2);
+      (*input) (r, c) = interpolate_color(-1.6f, 1.6f, z);
+    }
+
+  // filter
+  PointCloud<RGB>::Ptr output;
+  output.reset (new PointCloud<RGB> ());
+  pcl::filters::Convolution<RGB, RGB> convolve;
+  convolve.setInputCloud (input);
+  convolve.setKernel (filter);
+  convolve.convolveRows (*output);
+
+  // first output row
+  std::uint32_t i = 0;
+  for (std::uint32_t j = 0; j < output->width; ++j)
+  {
+    EXPECT_FLOAT_EQ((*output) (i, j).r, 0);
+    EXPECT_FLOAT_EQ((*output) (i, j).g, 0);
+    EXPECT_FLOAT_EQ((*output) (i, j).b, 0);
+  }
+
+  // last output row
+  i = output->height - 1;
+  for (std::uint32_t j = 0; j < output->width; ++j)
+  {
+    EXPECT_FLOAT_EQ((*output) (i, j).r, 0);
+    EXPECT_FLOAT_EQ((*output) (i, j).g, 0);
+    EXPECT_FLOAT_EQ((*output) (i, j).b, 0);
+  }
+}
+
+/// \todo Correct input values and filter factors required to test the function and not just detect compiler errors.
+TEST (Convolution, convolveRowsXYZRGB)
+{
+  // input
+  Eigen::ArrayXf filter(7);
+  filter << 0.00443305, 0.0540056, 0.242036, 0.39905, 0.242036, 0.0540056, 0.00443305;
+  PointCloud<PointXYZRGB>::Ptr input (new PointCloud<PointXYZRGB> ());
+  input->width = 48;
+  input->height = 48;
+  input->resize(input->width * input->height);
+  for (std::uint32_t r = 0; r < input->height; r++)
+    for (std::uint32_t c = 0; c < input->width; c++)
+    {
+      float x1 = -2.0f + (4.0f / (float)input->width) * (float)c;
+      float y1 = -2.0f + (4.0f / (float)input->height) * (float)r;
+      float x2 = -M_PI + (2.0f * M_PI / (float)input->width) * (float)c;
+      float y2 = -2.0f + (4.0f / (float)input->height) * (float)r;
+      float z = x1 * exp(-(x1 * x1 + y1 * y1)) * 2.5f + sin(x2) * sin(y2);
+      RGB color = interpolate_color(-1.6f, 1.6f, z);
+      (*input) (r, c).x = x1;
+      (*input) (r, c).y = y1;
+      (*input) (r, c).z = z;
+      (*input) (r, c).r = (uint8_t)(255.0f * color.r);
+      (*input) (r, c).g = (uint8_t)(255.0f * color.g);
+      (*input) (r, c).b = (uint8_t)(255.0f * color.b);
+    }
+
+  // filter
+  PointCloud<PointXYZRGB>::Ptr output;
+  output.reset (new PointCloud<PointXYZRGB> ());
+  pcl::filters::Convolution<PointXYZRGB, PointXYZRGB> convolve;
+  convolve.setInputCloud (input);
+  convolve.setKernel (filter);
+  convolve.convolveRows (*output);
+
+  // first output row
```

Comment:
Please add a note that this is because of border policy ignore which sets the border of width = filter_size/2 as NaN for xyz and 0 for rgb values.

Reply:
Ok